modules/apps/commerce/commerce-wish-list-test/src/testIntegration/java/com/liferay/commerce/wish/list/service/test/CommerceWishListItemLocalServiceTest.java

```diff
@@ -0,0 +1,106 @@
+import com.liferay.commerce.product.test.util.CPTestUtil;
+import com.liferay.commerce.wish.list.model.CommerceWishList;
+import com.liferay.commerce.wish.list.model.CommerceWishListItem;
+import com.liferay.commerce.wish.list.service.CommerceWishListItemLocalService;
+import com.liferay.commerce.wish.list.service.CommerceWishListService;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.rule.SynchronousDestinationTestRule;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Crescenzo Rega
+ */
+@RunWith(Arquillian.class)
+public class CommerceWishListItemLocalServiceTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE,
+			SynchronousDestinationTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+		_user = UserTestUtil.addUser();
+
+		_cpInstance = CPTestUtil.addCPInstance(_group.getGroupId());
+	}
+
+	@Test
+	public void testAddOrUpdateCommerceWishListItem() throws Exception {
+		_commerceWishList = _commerceWishListService.addCommerceWishList(
+			_group.getCompanyId(), RandomTestUtil.randomString(), true);
+
+		CPDefinition cpDefinition = _cpInstance.getCPDefinition();
+
+		CProduct cProduct = cpDefinition.getCProduct();
+
+		_commerceWishListItem =
+			_commerceWishListItemLocalService.addOrUpdateCommerceWishListItem(
+				_user.getUserId(), _commerceWishList.getCommerceWishListId(),
+				_cpInstance.getCPInstanceUuid(), cProduct.getCProductId(), "");
+
+		Assert.assertEquals(
+			1,
+			_commerceWishListItemLocalService.getCommerceWishListItemsCount(
+				_commerceWishList.getCommerceWishListId()));
+
+		_commerceWishListItemLocalService.addOrUpdateCommerceWishListItem(
+			_user.getUserId(), _commerceWishList.getCommerceWishListId(),
+			_cpInstance.getCPInstanceUuid(), cProduct.getCProductId(), "");
+
+		Assert.assertEquals(
+			1,
+			_commerceWishListItemLocalService.getCommerceWishListItemsCount(
+				_commerceWishList.getCommerceWishListId()));
+	}
+
+	private static User _user;
+
+	@DeleteAfterTestRun
+	private CommerceWishList _commerceWishList;
+
+	@DeleteAfterTestRun
+	private CommerceWishListItem _commerceWishListItem;
```

Comment:
I think these can be local variable because the data guard already delete them at the end of the test